Scenes/shop/shop_card.gd

```diff
@@ -0,0 +1,53 @@
+class_name ShopCard
+extends VBoxContainer
+
+const CARD_MENU_UI = preload("res://Scenes/ui/card_menu_ui.tscn")
+
+@export var card: Card:
+	set = set_card
+
+@onready var card_container: CenterContainer = %CardContainer
+@onready var price: HBoxContainer = %Price
+@onready var price_label: Label = %PriceLabel
+@onready var buy_button: Button = %BuyButton
+@onready var gold_cost := randi_range(100, 300)
+###Determine $ with fixed gold costs to all cards, a range, etc.
+
+@onready var current_card_ui: CardMenuUI
+#If I remove @onready "Definition out of order in None"
```

Comment:
Not sure i totally understand this comment but I think you're getting a linter error that it's out of order if you do var current_card_ui without the @onready. That's fine, you can just move it above the @onreadys.

Reply:
Yep, that fixed it!